dora/core/common/src/main/java/alluxio/membership/ServiceDiscoveryRecipe.java

```diff
@@ -60,57 +61,69 @@ public class ServiceDiscoveryRecipe {
   final AlluxioEtcdClient mAlluxioEtcdClient;
   private final ScheduledExecutorService mExecutor;
   private final String mClusterIdentifier;
-  // Will look like /ServiceDiscovery/<mClusterIdentifier>
-  private final String mRegisterPathPrefix;
-  private final ConcurrentHashMap<String, ServiceEntity> mRegisteredServices
+  // e.g. /ServiceDiscovery/<mClusterIdentifier>
+  private final Supplier<String> mRegisterPathPrefix
+      = CommonUtils.memoize(this::constructRegisterPathPrefix);
+  private final ConcurrentHashMap<String, DefaultServiceEntity> mRegisteredServices
       = new ConcurrentHashMap<>();
 
   /**
    * CTOR for ServiceDiscoveryRecipe.
    * @param client
    * @param clusterIdentifier
    */
-  public ServiceDiscoveryRecipe(AlluxioEtcdClient client, String clusterIdentifier) {
+  ServiceDiscoveryRecipe(AlluxioEtcdClient client, String clusterIdentifier) {
```

Comment:
license engine may need to construct its own `ServiceDiscoveryRecipe`. is there any context on this change?